### PR TITLE
Set TriggerDagRunOperator's trigger_rule to all_done in master_dag

### PR DESCRIPTION
**Pull request by pankajkoti**
The `master_dag` has implemented chaining of DAGs by grouping them based on providers. However, if one of the DAG fails in the chain, subsequent DAGs are not run and hence we do not test them. As such, the DAGs do not need to depend on each other and hence we can set the `trigger_rule` in the `TriggerDagRunOperator` to `all_done` so that we test all the DAGs irrespective of whether the previous DAG in the chain succeeded.

closes: #949

### CODE REVIEW

**Comment by pankajkoti** [comment]
The `mypy` failures will be resolved once PR https://github.com/astronomer/astronomer-providers/pull/951 is merged and this branch is then rebased on the main.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -96,6 +96,7 @@ repos:
         # Excluding Example DAGs temporarily until they can be fixed
         exclude: ^tests/.*\.py$|.*example_dags/.*
         args: ["-c", "./pyproject.toml"]
+        additional_dependencies: [".[toml]"]
```
Comment (phanikumv) [inline .pre-commit-config.yaml:99]: why this change?

Comment (pankajkoti) [reply .pre-commit-config.yaml:99]: Somehow, I could no longer commit without this. I was getting this error:
```
pyproject.toml : toml parser not available, reinstall with toml extra
```

I followed the Stackoverflow answer https://stackoverflow.com/questions/75303927/pyproject-toml-toml-parser-not-available-reinstall-with-toml-extra

Comment (pankajastro) [reply .pre-commit-config.yaml:99]: hmm, docs suggest something like https://bandit.readthedocs.io/en/latest/config.html?highlight=additional_dependencies#bandit-settings
```
additional_dependencies: ["bandit[toml]"]
```
not sure if it will work in this case

**Review by pankajastro — APPROVED** [review]

**Review by phanikumv — APPROVED** [review]
